From 3271eb4fc4bebd2e46d317a38771025f04c5edcd Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 13 Nov 2023 20:21:42 -0600
Subject: [PATCH] Use libxcrypt

Solus is using yescrypt as the default password hashing algorithm. Previously, the Plasma users KCM was creating hashes manually when updating 
user passwords. These hashes were sha512 which would then get saved to /etc/shadow by accountsservice. Let's use libxcrypt instead for salt generation
and hash generation. This will use yescrypt by default.

---
 kcms/users/src/user.cpp | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/kcms/users/src/user.cpp b/kcms/users/src/user.cpp
index 7949607fa..9ab02cf57 100644
--- a/kcms/users/src/user.cpp
+++ b/kcms/users/src/user.cpp
@@ -15,6 +15,7 @@
 #include <QtConcurrent>
 #include <sys/types.h>
 #include <unistd.h>
+#include <crypt.h>
 
 User::User(QObject *parent)
     : QObject(parent)
@@ -213,15 +214,7 @@ static char saltCharacter()
 
 static QString saltPassword(const QString &plain)
 {
-    QString salt;
-
-    salt.append("$6$");
-
-    for (auto i = 0; i < 16; i++) {
-        salt.append(saltCharacter());
-    }
-
-    salt.append("$");
+    QString salt = crypt_gensalt (NULL, 0, NULL, 0);
 
     auto stdStrPlain = plain.toStdString();
     auto cStrPlain = stdStrPlain.c_str();
-- 
2.42.0

